src/include/transaction/transaction_manager.h

```diff
@@ -76,6 +76,19 @@ class TransactionManager {
    */
   TransactionQueue CompletedTransactionsForGC();
 
+  /**
+   * Set the current buffer pool size
+   * @param new_size the new size of buffer pool
+   * @return true if the value is set successfully; false otherwise
+   */
+  bool SetBufferPoolSizeLimit(uint64_t new_size) { return buffer_pool_->SetSizeLimit(new_size); }
```

Comment:
The transaction manager holds a reference to the buffer segment pool, but I'm not convinced it actually owns it, and thus maybe isn't where these methods should reside.